google/cloud/bigquery_storage_v1/reader.py

```diff
@@ -149,15 +156,43 @@ def __iter__(self):
                 )
                 if not resumable_error:
                     raise
+            except google.api_core.exceptions.ResourceExhausted as exc:
+                # ResourceExhausted errors are only retried if a valid
+                # RetryInfo is provided with the error.
+                resumable_error = False
+                # ResourceExhausted doesn't seem to have
+                # details/_details fields by default when it is
+                # generated by Python 3.6 unit tests, so we have to
+                # work around that.
+                details = None
+                if hasattr(exc, "details"):
+                    details = exc.details
+                elif hasattr(exc, "_details"):
+                    details = exc._details
+                if details is not None:
+                    for detail in details:
+                        if isinstance(detail, google.rpc.error_details_pb2.RetryInfo):
+                            retry_delay = detail.retry_delay
+                            if retry_delay is not None:
+                                resumable_error = True
+                                delay = max(
+                                    0,
+                                    float(retry_delay.seconds)
+                                    + (float(retry_delay.nanos) / 1e9),
+                                )
+                                if self._retry_delay_callback:
+                                    self._retry_delay_callback(delay)
+                                time.sleep(delay)
+                            break
+                if not resumable_error:
+                    raise
             except _STREAM_RESUMPTION_EXCEPTIONS:
                 # Transient error, so reconnect to the stream.
                 pass
 
-            self._reconnect()
-
     def _reconnect(self):
         """Reconnect to the ReadRows stream using the most recent offset."""
-        self._wrapped = self._client.read_rows(
+        return self._client.read_rows(
```

Comment:
Won't this create a new ReadRowsStream?

Reply:
Yes, this function doesn't do anything different than before. It just returns the stream instead of assigning to a member variable.